### PR TITLE
feat: front-end changes to allow gateway renaming

### PR DESCRIPTION
**Pull request by tjvantoll**
# Problem Context

We want to allow users to change the name of their gateways.

## Changes

This PR introduces a new UI on the gateway details page that allows users to update their gateway names inline.

## Screenshot (if applicable)

gateway-rename

## Testing

* You can test this manually by visiting the gateway details page and using the new UI.
* There are also new unit and Cypress tests.

## Any other related PRs

Backend APIs (already merged to `main`): https://github.com/blues/sparrow-reference-web-app/pull/70

## Ticket(s)

https://www.pivotaltracker.com/story/show/181290351

### CODE REVIEW

File: src/pages/api/gateway/[gatewayUID]/name.ts
Comment (CarltonHenderson) [inline src/pages/api/gateway/[gatewayUID]/name.ts:30]: To delete this warning, join the [conversation](https://bluesinc.slack.com/archives/C02LAQLMJR5/p1646952266854589) on data validation.

File: src/styles/Details.module.scss
Comment (CarltonHenderson) [inline src/styles/Details.module.scss:10]: Can we encapsulate the edit-in-place stuff and include this too?

**Review by CarltonHenderson — APPROVED** [review]
Approved w/comments. Well done.

File: __tests__/src/components/elements/GatewayDetails.test.tsx
Comment (tjvantoll) [reply __tests__/src/components/elements/GatewayDetails.test.tsx:83]: The logic actually isn’t in the React component anymore, but I never moved this test. I’ll create a `gatewayDetails.test.ts` since that’s where the actual logic is now.

File: __tests__/src/components/elements/GatewayDetails.test.tsx
Comment (tjvantoll) [reply __tests__/src/components/elements/GatewayDetails.test.tsx:94]: It would definitely clean things up. I think I’ll hold off on this PR to avoid touching too much, and if we did that I’d kind of want to do all pages at once.

File: __tests__/src/components/elements/GatewayDetails.test.tsx
Comment (tjvantoll) [reply __tests__/src/components/elements/GatewayDetails.test.tsx:106]: Personally I’m ok with the bit of verbosity here.

File: cypress/integration/homepage.spec.js
Comment (tjvantoll) [reply cypress/integration/homepage.spec.js:165]: 👍

**Comment by tjvantoll** [comment]
@CarltonHenderson I think I addressed all your comments. When you have a minute could you take a quick look at `<EditInPlace>`? ([Here’s the code.](https://github.com/blues/sparrow-reference-web-app/blob/tj-gateway-rename-frontend/src/components/elements/EditInPlace.tsx)) Just want a sanity check on the API before I merge this.

**Comment by CarltonHenderson** [comment]
LGTM :+1:

**Pull request merged**
